src/k3s/mod.rs

```diff
@@ -0,0 +1,160 @@
+use std::collections::HashMap;
+use std::io;
+use std::io::ErrorKind;
+use std::path::{Path, PathBuf};
+
+use testcontainers::core::{Mount, WaitFor};
+use testcontainers::{Image, ImageArgs};
+
+const NAME: &str = "rancher/k3s";
+const TAG: &str = "v1.28.8-k3s1";
+pub const TRAEFIK_HTTP: u16 = 80;
+pub const KUBE_SECURE_PORT: u16 = 6443;
+pub const RANCHER_WEBHOOK_PORT: u16 = 8443;
+
+#[derive(Debug, Default, Clone)]
+pub struct K3s {
+    env_vars: HashMap<String, String>,
+    conf_mount: Option<Mount>,
+}
+
+#[derive(Default, Debug, Clone)]
+pub struct K3sArgs;
```

Comment:
I'd suggest to make it not-exhaustive, to be able to extend in the future without breaking changes.
E.g:
```rust
pub struct K3sArgs {
    _priv: (),
}

```

Reply:
I added the _priv field to make the struct non-exaustive